src/store/postgres/builder.rs

```diff
@@ -14,6 +15,27 @@ use crate::Aggregate;
 use super::persistable::Persistable;
 use super::{PgStore, Schema};
 
+/// The `EventIdUuidFormat` enum defines the UUID format of the event ID:
+///
+/// - `V4`: Uses the random UUID version 4 as defined by RFC 9562 section 5.4.
+/// - `V7`: Uses the time-ordered UUID version 7 as defined by RFC 9562 section 5.7.
+/// - `Custom`: Accepts a function used to generate UUIDs.
+pub enum EventIdUuidFormat {
+    V4,
+    V7,
+    Custom(fn() -> Uuid),
+}
+
+impl Into<fn() -> Uuid> for EventIdUuidFormat {
+    fn into(self) -> fn() -> Uuid {
+        match self {
+            EventIdUuidFormat::V4 => Uuid::new_v4,
+            EventIdUuidFormat::V7 => Uuid::now_v7,
+            EventIdUuidFormat::Custom(func) => func,
```

Comment:
My concern with allowing for a generic generator function with no parameters is that it is going to hook into the global scope for RNG (or even worse, generate a constant value) - i.e. I feel like it would be a dangerous footgun.

Reply:
This seems like something you would probably catch easily in testing but I'll remove it since I only need the V7 option.